### DISCUSSION TITLE
Legacy apps authentication?

### DISCUSSION THREAD

**Discussion by marcosimioni** [gravitational/teleport#12105] [Q&A] [answered]
Hello,
we use Teleport to provide access to a bunch of web apps.
Some of these web applications have their own login page and use LDAP to authenticate.
Other applications have their own login page with local authentication.

What is the recommended approach so that users don't have to login multiple times - first to Teleport, and then once for every web application? That's not great for the user experience.

Should I investigate into what and how authentication headers are used by the applications. and then try to forge them via `Headers passthrough` as described here https://goteleport.com/docs/application-access/guides/connecting-apps/#headers-passthrough ?

**Comment by webvictim** [comment, marked as answer]
The best supported way to do this is to use apps which have JWT support, or can add it via a plugin. Many apps like Grafana and Jenkins can do exactly this, and that allows Teleport to pass authentication information directly through to them. See the documentation here: https://goteleport.com/docs/application-access/guides/jwt/

If no JWT plugins are available, you might try injecting authentication headers in the way you described - although this is less secure.

**Reply by marcosimioni** [reply]
Thank you @webvictim ! I will look into that.
